source/source/deengames/planetoid/system/MouseShootSystem.hx

```diff
@@ -0,0 +1,61 @@
+package deengames.planetoid.system;
+
+import flixel.util.FlxColor;
+import flixel.FlxSprite;
+import flixel.FlxState;
+
+import flixel.addons.weapon.FlxWeapon;
+import flixel.addons.weapon.FlxBullet;
+import flixel.util.helpers.FlxBounds;
+import flixel.math.FlxPoint;
+
+import deengames.planetoid.component.MouseShootComponent;
+
+import nebula.ecs.component.MouseClickComponent;
+import nebula.ecs.component.SpriteComponent;
+import nebula.ecs.system.AbstractSystem;
+import nebula.ecs.Entity;
+
+class MouseShootSystem extends AbstractSystem
+{
+    private var state:FlxState;
+    
+    public function new(state:FlxState)
+    {
+        super([MouseShootComponent, MouseClickComponent, SpriteComponent]);
+        this.state = state;
+    }
+    
+    
+    override public function entityChanged(entity:Entity):Void
+    {
+        super.entityChanged(entity);
+
+        for (entity in this.entities)
+        {
+            var mouseShootComponent:MouseShootComponent = this.entities[0].get(MouseShootComponent);
+            var spriteComponent:SpriteComponent = this.entities[0].get(SpriteComponent);
+            var sprite = spriteComponent.sprite;
+            if (mouseShootComponent.gun == null)
+            {
+                mouseShootComponent.gun = new FlxWeapon("gun",createNormalGunBullet,
+                                FlxWeaponFireFrom.PARENT(spriteComponent.sprite, new FlxBounds(new FlxPoint(sprite.width / 2, sprite.height / 2),
+                                                                                                 new FlxPoint(sprite.width / 2, sprite.height / 2))),
+                                FlxWeaponSpeedMode.SPEED(new FlxBounds(150.0, 200)));
+                mouseShootComponent.gun.fireRate = mouseShootComponent.fireRate;
+                mouseShootComponent.gun.bulletLifeSpan = mouseShootComponent.bulletLifeSpan;
+                mouseShootComponent.gun.rotateBulletTowardsTarget = true;
+                
+                //TODO : find a way to set map bounds
```

Comment:
there maybe a problem when the character moves out of the stage. because bullets get destroyed automatically then. so we need to have some sort of bounds.